From cc809a01ac3059fa613473ce45bc75426ab000ad Mon Sep 17 00:00:00 2001
From: Peter Rajnoha <prajnoha@redhat.com>
Date: Wed, 23 Jul 2025 09:17:36 +0200
Subject: [PATCH 15/47] device_id: log_warn instead of log_error in
 device_ids_write

It is not always the case that the whole command fails if
device_ids_write fail so change all log_error to log_warn and let the
caller decide.

(cherry picked from commit 80144223e700703f76267b8d033e2d7d04156ee9)
---
 lib/device/device_id.c | 20 ++++++++++----------
 1 file changed, 10 insertions(+), 10 deletions(-)

diff --git a/lib/device/device_id.c b/lib/device/device_id.c
index 76165559f..4bb92cb09 100644
--- a/lib/device/device_id.c
+++ b/lib/device/device_id.c
@@ -1730,12 +1730,12 @@ int device_ids_write(struct cmd_context *cmd)
 		log_sys_debug("unlink", tmppath);
 
 	if (!(fp = fopen(tmppath, "w+"))) {
-		log_error("Cannot open to write %s.", tmppath);
+		log_warn("WARNING: %s: cannot open for writing: %s.", tmppath, strerror(errno));
 		goto out;
 	}
 
 	if ((dir_fd = open(dirpath, O_RDONLY)) < 0) {
-		log_sys_error("open", dirpath);
+		log_warn("WARNING: %s: cannot open directory: %s.", dirpath, strerror(errno));
 		goto out;
 	}
 
@@ -1748,7 +1748,7 @@ int device_ids_write(struct cmd_context *cmd)
 	fb_size = names_len + 256 + (128 * dm_list_size(&cmd->use_devices));
 
 	if (!(fb = malloc(fb_size))) {
-		log_error("Failed to allocate buffer size %d for devices file.", fb_size);
+		log_warn("Failed to allocate buffer size %d for devices file.", fb_size);
 		goto out;
 	}
 
@@ -1762,7 +1762,7 @@ int device_ids_write(struct cmd_context *cmd)
 		num = snprintf(fb + pos, len - pos, "HOSTNAME=%s\n", cmd->hostname);
 
 	if (num >= len - pos) {
-		log_error("Failed to write buffer for devices file content.");
+		log_warn("Failed to write buffer for devices file content.");
 		goto out;
 	}
 	pos += num;
@@ -1772,7 +1772,7 @@ int device_ids_write(struct cmd_context *cmd)
 
 	num = snprintf(fb + pos, len - pos, "%s\n", version_buf);
 	if (num >= len - pos) {
-		log_error("Failed to write buffer for devices file content.");
+		log_warn("Failed to write buffer for devices file content.");
 		goto out;
 	}
 	pos += num;
@@ -1801,7 +1801,7 @@ int device_ids_write(struct cmd_context *cmd)
 					du->idname ?: ".", devname, pvid);
 		}
 		if (num >= len - pos) {
-			log_error("Failed to write buffer for devices file content.");
+			log_warn("Failed to write buffer for devices file content.");
 			goto out;
 		}
 		pos += num;
@@ -1820,17 +1820,17 @@ int device_ids_write(struct cmd_context *cmd)
 			    "# HASH=%u\n",
 			    cmd->name, cmd->device_ids_auto_import ? " (auto)" : "",
 			    getpid(), ctime(&t), hash)) < 0) {
-		log_error("Failed to write buffer for devices file content.");
+		log_warn("Failed to write buffer for devices file content.");
 		goto out;
 	}
 	fc[fc_bytes] = '\0';
 
 	if (fputs(fc, fp) < 0) {
-		log_error("Failed to write devices file header.");
+		log_warn("Failed to write devices file header.");
 		goto out;
 	}
 	if (fputs(fb, fp) < 0) {
-		log_error("Failed to write devices file.");
+		log_warn("Failed to write devices file.");
 		goto out;
 	}
 	if (fflush(fp) < 0)
@@ -1845,7 +1845,7 @@ int device_ids_write(struct cmd_context *cmd)
 	fp = NULL;
 
 	if (rename(tmppath, cmd->devices_file_path) < 0) {
-		log_error("Failed to replace devices file.");
+		log_warn("WARNING: %s: failed to replace devices file: %s", cmd->devices_file_path, strerror(errno));
 		goto out;
 	}
 
-- 
2.51.0

